static/js/upload.js

```diff
@@ -53,21 +54,34 @@ exports.options = function (config) {
         send_button.attr("disabled", "");
         send_status.addClass("alert-info").show();
         send_status_close.one('click', compose.abort_xhr);
-        error_msg.html($("<p>").text(i18n.t("Uploading…"))
-            .after('<div class="progress progress-striped active">' +
-                   '<div class="bar" id="' + upload_bar + '" style="width: 00%;"></div>' +
-                   '</div>'));
+        error_msg.html($("<p>").text(i18n.t("Uploading…")));
+        send_status.append('<div class="progress progress-striped active">' +
+                           '<div class="bar" id="' + upload_bar + '" style="width: 00%;"></div>' +
+                           '</div>');
+        upload_is_finished = false;
     };
 
     var progressUpdated = function (i, file, progress) {
-        $("#" + upload_bar).width(progress + "%");
+        $("#" + upload_bar).animate({
+            width: progress + "%",
+        }, 500, "linear", function () {
```

Comment:
This doesn't work well -- if you set network to 3G in Chrome development tools and upload a  1MB file, you'll see the progress bar oscillating up and down.